src/components/SeekGraph/SeekGraphLegend.tsx

```diff
@@ -0,0 +1,192 @@
+            <>
+                {icon}
+                <input
+                    id={text}
+                    type="checkbox"
+                    checked={props.filter[filterKey]}
+                    onChange={() => props.toggleHandler(filterKey)}
+                ></input>
+                <label htmlFor={text}>{text}</label>
+            </>
+        );
+    };
+
+    const legendItemNoToggle = (text: string, icon: JSX.Element): JSX.Element => {
+        return (
+            <>
+                {icon}
+                <label htmlFor={text}>{text}</label>
+            </>
+        );
+    };
+
+    const leftColumn = [
+        legendItem(_("19x19"), BoardSizeLegendIcon(currentPalette.size19), "show19x19"),
+        legendItem(_("13x13"), BoardSizeLegendIcon(currentPalette.size13), "show13x13"),
+        legendItem(_("9x9"), BoardSizeLegendIcon(currentPalette.size9), "show9x9"),
+        legendItem(_("Other"), BoardSizeLegendIcon(currentPalette.sizeOther), "showOtherSizes"),
+        legendItemNoToggle(_("My challenges"), BoardSizeLegendIcon(currentPalette.user)),
+    ];
+    const rightColumn = [
+        legendItem(
+            _("Ranked"),
+            LegendIcon((ctx) =>
+                drawChallengeSquare(
+                    ICON_CENTER.x,
+                    ICON_CENTER.y,
+                    ICON_HEIGHT - 2,
+                    currentPalette.legend,
+                    ctx,
+                ),
+            ),
+            "showRanked",
+        ),
+        legendItem(
+            _("Unranked"),
+            LegendIcon((ctx) =>
+                drawChallengeTriangle(
+                    ICON_CENTER.x,
+                    ICON_CENTER.y,
+                    ICON_HEIGHT - 2,
+                    currentPalette.legend,
+                    ctx,
+                ),
+            ),
+            "showUnranked",
+        ),
+        legendItem(
+            _("Rengo"),
+            LegendIcon((ctx) =>
+                drawChallengeCircle(
+                    ICON_CENTER.x,
+                    ICON_CENTER.y,
+                    (ICON_HEIGHT - 2) / 2,
+                    currentPalette.legend,
+                    ctx,
+                ),
+            ),
+            "showRengo",
+        ),
+        legendItem(
+            _("Ineligible"),
+            BoardSizeLegendIcon(currentPalette.ineligible),
+            "showIneligible",
+        ),
+        legendItemNoToggle(
+            _("My rank"),
+            LegendIcon((ctx) => {
+                ctx.beginPath();
+                ctx.moveTo(0, ICON_CENTER.y);
+                ctx.lineTo(ICON_WIDTH, ICON_CENTER.y);
+                ctx.strokeStyle = currentPalette.rankLineColor;
+                ctx.stroke();
+            }),
+        ),
+    ];
+
+    return (
+        <div className="seek-graph-legend">
+            <Card>
+                <table>
+                    <tbody>
+                        {[0, 1, 2, 3, 4].map((i) => (
+                            <tr>
```

Comment:
Here's that key think I was talking about. ([React Docs: Lists and Keys](https://reactjs.org/docs/lists-and-keys.html))

```suggestion
                            <tr key={i}>
```

Reply:
Thanks for the reminder! I’d actually meant to fix that but it completely slipped my mind.

Reply:
The SeekGraph issue is probably entirely unrelated – it’s a different component than the legend. What happens when you inspect the area where it should show up?

Reply:
Never mind, fixed. Firefox doesn’t support a certain font measure I was using, which caused a padding value to be NaN, which cascaded and made all rendering coordinates NaN…

Reply:
yep the keys thing was def unrelated haha, just the first thing i saw in the console when I hit inspect.

Thanks for figuring out the Seek Graph issue - that was a strange root cause!